### PR TITLE
chore: replace hmpps date picker with moj one

### PR DESCRIPTION
**Pull request by ReedSoftware**

### CODE REVIEW

File: server/journeys/referral/details/view.njk [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {% from "govuk/components/date-input/macro.njk" import govukDateInput %}
 {% from "govuk/components/select/macro.njk" import govukSelect %}
 {% from "govuk/components/button/macro.njk" import govukButton %}
-{% from "hmpps/components/date-picker/macro.njk" import hmppsDatePicker %}
+{% from "moj/components/date-picker/macro.njk" import mojDatePicker %}
```
Comment (bill-moj) [inline server/journeys/referral/details/view.njk:4]: the js import below:
```
{% block additionalScripts %}
    <script src="/assets/javascript/datepicker.js"></script>
{% endblock %}
```
was added to support `hmppsDatePicker`
does `mojDatePicker` need something like that, or can this js import be removed?

Comment (iterative-it) [reply server/journeys/referral/details/view.njk:4]: Yes these, and the ones in the other three views the date picker is in can have this block removed

File: server/journeys/referral/details/schemas.ts [hunk 2]
```diff
@@ -33,15 +37,28 @@ export const schemaFactory =
...
-              ctx.addIssue({ code: z.ZodIssueCode.custom, message: INCIDENT_DATE_FUTURE_MSG })
-            }
+          .min(8, { message: INCIDENT_DATE_MSG })
+          .max(10, { message: INCIDENT_DATE_MSG })
+          .transform(value => value.split(/[-/]/).reverse())
+          .transform(value => {
+            // Prefix month and date with a 0 if needed
+            const month = value[1]?.length === 2 ? value[1] : `0${value[1]}`
+            const date = value[2]?.length === 2 ? value[2] : `0${value[2]}`
+            return `${value[0]}-${month}-${date}T00:00:00Z` // We put a full timestamp on it so it gets parsed as UTC time and the date doesn't get changed due to locale
```
Comment (iterative-it) [inline server/journeys/referral/details/schemas.ts:47]: Are there any existing tests for the date parsing?

File: server/journeys/referral/details/view.njk (hunk 1 above)
Comment (ReedSoftware) [reply server/journeys/referral/details/view.njk:4]: It's only in this one view that I can find?

File: server/journeys/referral/details/schemas.ts (hunk 2 above)
Comment (ReedSoftware) [reply server/journeys/referral/details/schemas.ts:47]: For the different edge cases? There are a couple in the tests.test.ts file, and I added more in the e2e PR I've done.

File: server/journeys/referral/details/view.njk (hunk 1 above)
Comment (iterative-it) [reply server/journeys/referral/details/view.njk:4]: I found these in the branch
image

Comment (ReedSoftware) [reply server/journeys/referral/details/view.njk:4]: I merged those two together as part of 334 here https://github.com/ministryofjustice/hmpps-challenge-support-intervention-plan-ui/pull/73/files

Comment (ReedSoftware) [reply server/journeys/referral/details/view.njk:4]: You might be looking at an old view of the code?

Comment (iterative-it) [reply server/journeys/referral/details/view.njk:4]: They were in /dist! Sorry

**Review by iterative-it — APPROVED** [review]
LGTM
